feature: Terraform setup for APIM management connection

lechnerc77 opened the pull request:
## Purpose

To get a feature parity between the Terraform and the Biceo setup this PR includes the changes necessary to conditionally deploy the APIM configuration to connect an the app to an S/4HANA backend with exitsing APIM fronting.


## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Set the infrastructure provisiong to Terraform in the `azure.yaml` file (this will lead to a syntax error in the yaml, which is fine):
  ```yaml 
   infra:
    provider: terraform
    path: ./infra-terraform
  ```
* Regression: deploy the  app to Azure via  `azd up`
* APIM deployment: switch the parameter  for the APIM provisioning to true (file `infra-terraform/variables.tf` -> `useAPIM` set the default value to true) and deploy the  app to Azure via  `azd up` 

## What to Check
Verify that the following are valid
* in both cases (regression and APIM deployment) the app is successfully deployed to Azure

## Other Information

n/a

lechnerc77 commented:
@MartinPankraz : Can you please check if this works in your setup with the APIM management as descripbed in the APIM variant of the PR?

MartinPankraz commented:
Not quite there yet: 

![image](https://github.com/Azure-Samples/app-service-javascript-sap-cloud-sdk-quickstart/assets/5108923/056ea853-c3f9-4952-b269-88e3dc6f4f09)

Putting "useAPIM" to false creates a successful run.

Maybe elevate azd version for that branch too?

lechnerc77 commented:
> Not quite there yet:
> 
> ![image](https://user-images.githubusercontent.com/5108923/242510998-056ea853-c3f9-4952-b269-88e3dc6f4f09.png)
> 
> Putting "useAPIM" to false creates a successful run.
> 
> Maybe elevate azd version for that branch too?

Would have been a surprise if this would work in the first run. Taking it step by step:

* I fixed the wrong parameter in the logger
* I removed the app properties to exclude the error form this resource.

Onece the logger works I will check and fix the web site config issue
